File: app/src/main/java/sfllhkhan95/doodle/core/utils/ActionBarManager.java
package sfllhkhan95.doodle.core.utils;

import android.view.Menu;
import android.view.MenuItem;

import sfllhkhan95.doodle.R;
import sfllhkhan95.doodle.core.views.PaintView;

/**
 * @author saifkhichi96
 * @see CanvasActionListener
 */
public class ActionBarManager implements CanvasActionListener {

    // Action bar buttons
    private final MenuItem undoAction;
    private final MenuItem redoAction;
    private final MenuItem revertAction;
    private final MenuItem saveAction;
    //private final MenuItem saveAsAction;

    public ActionBarManager(Menu menu) {
        undoAction = menu.findItem(R.id.undo);
        redoAction = menu.findItem(R.id.redo);
        revertAction = menu.findItem(R.id.revert);
        saveAction = menu.findItem(R.id.save);
        //saveAsAction = menu.findItem(R.id.save_as);
    }

    private void enable(MenuItem item) {
        item.setEnabled(true);
    }

    private void disable(MenuItem item) {
        item.setEnabled(false);
    }

    @Override
    public void onDrawPath() {
        enable(revertAction);
        enable(undoAction);
        disable(redoAction);
        enable(saveAction);
        //enable(saveAsAction);
    }

    @Override
    public void onUndo(boolean canUndo) {
        if (!canUndo) {
            disable(revertAction);
            disable(undoAction);
            disable(saveAction);
            //disable(saveAsAction);
        }

        enable(redoAction);
    }

    @Override
    public void onRedo(boolean canRedo) {
        if (!canRedo) {
            disable(redoAction);
        }

        enable(revertAction);
        enable(undoAction);
        enable(saveAction);
        //enable(saveAsAction);
    }

    @Override
    public void onRevert() {
        disable(revertAction);
        disable(undoAction);
        disable(redoAction);
        disable(saveAction);
        //disable(saveAsAction);
    }

    public void sync(PaintView paintView) {
        if (paintView.isModified()) {
            enable(revertAction);
            enable(undoAction);
            enable(saveAction);
            //enable(saveAsAction);
        } else {
            disable(revertAction);
            disable(undoAction);
            disable(saveAction);
            //disable(saveAsAction);
        }

        if (paintView.canRedo()) {
            enable(redoAction);
        } else {
            disable(redoAction);
        }
    }
}

File: app/src/main/java/sfllhkhan95/doodle/core/utils/ActionBarManager.kt
package sfllhkhan95.doodle.core.utils

import android.view.Menu
import android.view.MenuItem

import sfllhkhan95.doodle.R
import sfllhkhan95.doodle.core.views.PaintView

/**
 * @author saifkhichi96
 * @see CanvasActionListener
 */
class ActionBarManager
(menu: Menu) : CanvasActionListener {

    // Action bar buttons
    private val undoAction: MenuItem = menu.findItem(R.id.undo)
    private val redoAction: MenuItem = menu.findItem(R.id.redo)
    private val revertAction: MenuItem = menu.findItem(R.id.revert)
    private val saveAction: MenuItem = menu.findItem(R.id.save)
    // private val saveAsAction: MenuItem = menu.findItem(R.id.save_as)

    init {

    }

    private fun enable(item: MenuItem) {
        item.isEnabled = true
    }

    private fun disable(item: MenuItem) {
        item.isEnabled = false
    }

    override fun onDrawPath() {
        enable(revertAction)
        enable(undoAction)
        disable(redoAction)
        enable(saveAction)
        //enable(saveAsAction);
    }

    override fun onUndo(allowed: Boolean) {
        if (!allowed) {
            disable(revertAction)
            disable(undoAction)
            disable(saveAction)
            //disable(saveAsAction);
        }

        enable(redoAction)
    }

    override fun onRedo(allowed: Boolean) {
        if (!allowed) {
            disable(redoAction)
        }

        enable(revertAction)
        enable(undoAction)
        enable(saveAction)
        //enable(saveAsAction);
    }

    override fun onRevert() {
        disable(revertAction)
        disable(undoAction)
        disable(redoAction)
        disable(saveAction)
        //disable(saveAsAction);
    }

    fun sync(paintView: PaintView) {
        if (paintView.isModified) {
            enable(revertAction)
            enable(undoAction)
            enable(saveAction)
            //enable(saveAsAction);
        } else {
            disable(revertAction)
            disable(undoAction)
            disable(saveAction)
            //disable(saveAsAction);
        }

        if (paintView.canRedo()) {
            enable(redoAction)
        } else {
            disable(redoAction)
        }
    }
}
